mongodb reactive

sdelamo opened the pull request.

sdelamo committed:
mongodb reactive

ilopmar commented on guides/micronaut-mongodb-asynchronous/java/src/main/java/example/micronaut/MongoDbFruitRepository.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    @Override
+    public Mono<Boolean> save(@NonNull @NotNull @Valid Fruit fruit){
+        return Mono.from(getCollection().insertOne(fruit)) // <4>
+                .map(insertOneResult -> Boolean.TRUE)
+                .onErrorReturn(Boolean.FALSE);
+    }
```
Same comment as in the Mongo Sync guide. I think it is a common use case to have an `id` field assigned automatically Mongo and then return the `Fruit` object with that `id` field updated.

graemerocher commented on guides/micronaut-mongodb-asynchronous/java/src/test/java/example/micronaut/FruitControllerTest.java:
```diff
@@ -0,0 +1,44 @@
+import java.util.Collections;
+import java.util.List;
+
+public class FruitControllerTest {
```
Does it make sense to share this template with the synchronous guide since they should be the same

sdelamo replied:
We don't have support for sharing code between guides.

ilopmar reviewed (commented):
One additional comment. Wouldn't it be better if you merge both guides into a single one and explain how to use both blocking and reactive clients?

graemerocher commented:
I personally kind of like that they are separate to explain best practise for each

sdelamo commented:
I prefer them to be separate. Shorter guides for the win.

sdelamo committed:
copy changes

See https://github.com/micronaut-projects/micronaut-guides/pull/528/commits/51b3df5d8af7ec506a37fd9acc9d1e33295c9007

sdelamo committed:
change sentence of connection

replace

The application connects to it thanks to the following configuration:

with:

The application connects to MongoDb, run via Docker, thanks to the following configuration:

sdelamo commented:
@kroegerk can you proofread it
